pkg/render/logstorage.go

```diff
@@ -837,15 +834,8 @@ func (es elasticsearchComponent) elasticsearchCluster(secureSettings bool) *esv1
 			},
 		},
 		Spec: esv1.ElasticsearchSpec{
-			Version: components.ComponentEckElasticsearch.Version,
-			Image:   es.esImage,
-			HTTP: cmnv1.HTTPConfig{
```

Comment:
This might warrant further discussion, are we removing BYO certs for Elasticsearch right now?

I'm not sure if this also affects certificate management. Taking this out means that the ECK operator will generate a public certificate that for Elasticsearch (based off the private certificate it creates and stores in a secret), and we can't change this (or else the ECK operator will change it back).

This means that certificate management will be broken, since it can't override the public certificate, or you'd have to mount the default secret generated by elasticsearch if cert management wasn't enabled, and a different one if it was, which makes this a very uncomfortable configuration situation.

We always need to be the one to generate the certificate, ECK will copy the one created here to the "default" secret name instead of generating it. All default certificates must be generated by us. Please either change this back, or generate a differently named certificate secret that we generate by default.

CC @rene-dekker, is there something I'm missing here?

Reply:
I think everything you say is essentially correct, @Brian-McM.
As for certificate management, there are many ways to get esgw to trust elasticsearch:
- Doing as Brian suggests would be the least number of code changes probably.
- Configure esgw to read es cert from /path/to/ca.crt from the init container (which is a bit hacky)
- Configure esgw to put  /path/to/ca.crt from the init container into the OS trusted cert directory.
I would be ok with any of them personally, but maybe for the sake of getting this PR ready, the first option would be a good choice.

Reply:
I'm not OK with that last two options until certificate management is done like that for all components. Currently cert management work by creating a public certificate secret from the private one created in the pod, so that's how it should stay for all components until a new design has been put forward to change this and has been approved.